pkg/cmd/get/get_preview_test.go

```diff
@@ -0,0 +1,160 @@
+package get_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/jenkins-x/jx/pkg/kube"
+
+	"github.com/jenkins-x/jx/pkg/cmd/get"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/testhelpers"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	jenkinsv1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	clientmocks "github.com/jenkins-x/jx/pkg/cmd/clients/mocks"
+	helm_test "github.com/jenkins-x/jx/pkg/helm/mocks"
+	resources_test "github.com/jenkins-x/jx/pkg/kube/resources/mocks"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCurrentPreviewUrlWithoutEnvVars(t *testing.T) {
+	ro := os.Getenv("REPO_OWNER")
+	rn := os.Getenv("REPO_NAME")
+	jn := os.Getenv("JOB_NAME")
+
+	defer os.Setenv("REPO_OWNER", ro)
+	defer os.Setenv("REPO_NAME", rn)
+	defer os.Setenv("JOB_NAME", jn)
+
+	commonOpts := &opts.CommonOptions{}
+	factory := clientmocks.NewMockFactory()
+	commonOpts.SetFactory(factory)
+
+	options := &get.GetPreviewOptions{
+		GetEnvOptions: get.GetEnvOptions{
+			GetOptions: get.GetOptions{
+				CommonOptions: commonOpts,
+			},
+		},
+	}
+
+	want := "No $JOB_NAME defined for the current pipeline job to use"
+
+	if _, err := options.CurrentPreviewUrl(); err != nil {
+		if err.Error() != want {
+			t.Errorf("CurrentPreviewUrl() failed with wrong error got `%s' want `%s'", err.Error(), want)
+		}
+	} else {
+		t.Errorf("CurrentPreviewUrl() didn't fail, expected error: `%s'", want)
+	}
+}
+
+func TestCurrentPreviewUrlWithoutPreviews(t *testing.T) {
+	os.Setenv("REPO_OWNER", "jx-testing")
+	os.Setenv("REPO_NAME", "jx-testing")
+	defer os.Unsetenv("REPO_OWNER")
+	defer os.Unsetenv("REPO_NAME")
+
+	commonOpts := &opts.CommonOptions{}
+	commonOpts.SetDevNamespace("jx-testing")
+
+	factory := clientmocks.NewMockFactory()
+	user := jenkinsv1.User{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "jx",
+			Name:      "test-user",
+		},
+		Spec: jenkinsv1.UserDetails{
+			Name:     "Test",
+			Email:    "test@test.com",
+			Accounts: make([]jenkinsv1.AccountReference, 0),
+		},
+	}
+
+	testhelpers.ConfigureTestOptionsWithResources(commonOpts,
```

Comment:
I think you need something like this in `TestCurrentPreviewUrlWithoutEnvVars` too.